### PR TITLE
zephyr: Allow single image applications

### PR DESCRIPTION
**Pull request by de-nordic**
The change enables use of single image applications, for those that do
not need to use multiple images on device.
Single image means that the update of application will overwrite image
of previous application.
image_0 partition (primary) is used for loading application with single
image configuration, all other image partitions are ignored.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -58,6 +58,14 @@ config NRF_CC310_BL
 
 menu "MCUBoot settings"
 
+config SINGLE_IMAGE_DFU
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig:61]: This is equivalent to `MCUBOOT_IMAGE_NUMBER == 0.5 true case`. Shouldn't be ensure that MCUBOOT_IMAGE_NUMBER is equal to 1?

Comment (de-nordic) [reply boot/zephyr/Kconfig:61]: It could be checked.

Comment (de-nordic) [reply boot/zephyr/Kconfig:61]: Do you want the check to be done here boot/zephyr/include/target.h, same way the other checks are done?

Comment (de-nordic) [reply boot/zephyr/Kconfig:61]: OK, done it via limiting range on `UPDATEABLE_IMAGE_NUMBER`, in Kconfig, when `SINGLE_IMAGE_DFU` is selected.

**Comment by nvlsianpu** [comment]
Looks very promising - will test this on Monday.

**Comment by utzig** [comment]
I tested this and it worked correctly for me. Needs rebase!

**Comment by de-nordic** [comment]
> 
> 
> I tested this and it worked correctly for me. Needs rebase!

Thanks!

**Pull request merged**